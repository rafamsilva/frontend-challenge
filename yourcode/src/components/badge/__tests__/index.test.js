import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import Badge from '../';
import { ReactComponent as ImdbLogo } from '../../../assets/images/logo-imdb.svg';

describe('Badge Component', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('should render without crashing', () => {
    act(() => {
      render(<Badge value="10" />, container);
    });
  });

  it('should render without an icon', () => {
    act(() => {
      render(<Badge value="10" />, container);
    });

    expect(container.querySelector("[data-testid='icon']")).toBeNull();
    expect(container.querySelector("[data-testid='text']").textContent).toBe('10');
  });

  it('should render with an icon', () => {
    act(() => {
      render(<Badge value="10" icon={<ImdbLogo alt="IMDB Logo" title="IMDB Logo" />} />, container);
    });

    expect(container.querySelector("[data-testid='icon']")).toBeTruthy();
    expect(container.querySelector("[data-testid='text']").textContent).toBe('10');
  });

  describe('Should match snapshot', () => {
    it('should match snapshot without icon', () => {
      act(() => {
        render(<Badge value="10" />, container);
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });

    it('should match snapshot with icon', () => {
      act(() => {
        render(<Badge value="10" icon={<ImdbLogo alt="IMDB Logo" title="IMDB Logo" />} />, container);
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
  });
});
