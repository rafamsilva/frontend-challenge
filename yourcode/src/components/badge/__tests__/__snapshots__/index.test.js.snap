// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component Snapshot should match snapshot with icon 1`] = `"<div class=\\"sc-AxjAm gFqxcz\\"><span data-testid=\\"icon\\" class=\\"sc-AxirZ dTmdqe\\"><svg alt=\\"IMDB Logo\\" title=\\"IMDB Logo\\">logo-imdb.svg</svg></span><span data-testid=\\"text\\" class=\\"sc-AxiKw jVCyOE\\">10</span></div>"`;

exports[`Badge Component Snapshot should match snapshot without icon 1`] = `"<div class=\\"sc-AxjAm gFqxcz\\"><span data-testid=\\"text\\" class=\\"sc-AxiKw jVCyOE\\">10</span></div>"`;
